modules/tinymce/src/core/test/ts/browser/EditableRootTest.ts

```diff
@@ -0,0 +1,97 @@
+import { context, describe, it } from '@ephox/bedrock-client';
+import { TinyHooks } from '@ephox/wrap-mcagar';
+import { assert } from 'chai';
+
+import Editor from 'tinymce/core/api/Editor';
+
+describe('browser.tinymce.core.EditableRootTest', () => {
+  const assertContentEditableState = (editor: Editor, expectedState: boolean) => {
+    assert.equal(editor.getBody().contentEditable === 'true', expectedState);
+  };
+
+  const assertEditableState = (editor: Editor, expectedState: boolean) => {
+    assert.equal(editor.hasEditableRoot(), expectedState);
+    assertContentEditableState(editor, expectedState);
+  };
+
+  context('editable_root: false', () => {
+    let initialContentEditableState = '';
+
+    const hook = TinyHooks.bddSetupLight<Editor>({
+      base_url: '/project/tinymce/js/tinymce',
+      editable_root: false,
+      setup: (editor: Editor) => {
+        editor.on('PreInit', () => {
+          initialContentEditableState = editor.getBody().contentEditable;
+        });
+      }
+    }, [], true);
+
+    it('TINY-9839: Should not have an editable root state or editable body', () => {
+      assertEditableState(hook.editor(), false);
+      assert.equal(initialContentEditableState, 'inherit', 'Should not be editable early in the init process');
+    });
+  });
+
+  context('editable_root: default', () => {
+    const hook = TinyHooks.bddSetupLight<Editor>({
+      base_url: '/project/tinymce/js/tinymce'
+    }, [], true);
+
```

Comment:
maybe we need to reset the state for the tests do not affect each other
`afterEach(() => {
    editor.setEditableRoot(true);
})`

Reply:
I think I reset the current test by setting back the state I guess we could use a after if resetting the state becomes a problem and if there is more than just this simple test.